### PR TITLE
Change on authentication: disabling open data and super user

### PR DESCRIPTION
**Pull request by kinnou02**
User as now a type that will be used for authentication, there is
currently three types of user:
- with_free_instances: same as before, have access to all free
  instance of kraken
- without_free_instances: this type of user don't have access to free
  instance if the authorization is not granted explicitly
- super_user: they have access to all instances free or not!

A new parameter has been added to tyr for handling this

### CODE REVIEW

File: source/jormungandr/tests/authentication_tests.py [hunk 1]
```diff
@@ -240,6 +259,17 @@ def test_stop_schedules_for_bobette(self):
                                              'stopbidon/stop_schedules')
             assert status == 404
 
+    def test_stop_schedules_for_tgv(self):
+        with user_set(app, 'tgv'):
+            response = self.query('v1/coverage/main_routing_test/stop_areas/stopA/stop_schedules?from_datetime=20120614T080000')
+            assert 'error' not in response
+            _, status = self.query_no_assert('v1/coverage/departure_board_test/stop_areas/stop1/stop_schedules')
+            eq_(status, 403)
+            #we get a 404 (because 'stopbidon' cannot be found) and not a 403
```
Comment (antoine-de) [inline source/jormungandr/tests/authentication_tests.py]: you need to update this comment

File: source/jormungandr/tests/authentication_tests.py [hunk 2]
```diff
@@ -56,6 +57,12 @@
         "departure_board_test": {'ALL': True},
         "empty_routing_test": {'ALL': False}
     },
+    'tgv': {
+        #bobitto can access all since empty_routing_test is free
```
Comment (antoine-de) [inline source/jormungandr/tests/authentication_tests.py]: you need to update this comment too

File: source/tyr/migrations/versions/55bf63d1c18_add_fields_enable_free_on_users.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('user', sa.Column('enable_free_instances', sa.Boolean(), nullable=False, server_default='True'))
+    op.add_column('user', sa.Column('is_super_user', sa.Boolean(), nullable=False, server_default='False'))
+    ### end Alembic commands ###
+
+
+def downgrade():
+    ### commands auto generated by Alembic - please adjust! ###
+    op.drop_column('user', 'is_super_user')
+    op.drop_column('user', 'enable_free_instances')
```
Comment (TeXitoi) [inline source/tyr/migrations/versions/55bf63d1c18_add_fields_enable_free_on_users.py]: Better to have 3 kinds of users: {with_free_instances, without_free_instances, super_user} ?

**Comment by antoine-de** [comment]
Jenkins, test this please

**Pull request merged**